pkg/chartutil/save.go

```diff
@@ -116,28 +117,50 @@ func Save(c *chart.Chart, outDir string) (string, error) {
 		return "", err
 	}
 
+	rollback := false
+	// save the chart to the file
+	err = Write(c, f)
+
+	if err != nil {
+		rollback = true
+		return filename, err
+	}
+
+	defer func() {
+		f.Close()
+		if rollback {
+			os.Remove(filename)
+		}
+	}()
+
+	return filename, nil
+}
+
+// Write streams an archived chart to an io.writer interface.
+//
+// This takes an existing chart and a destination writer.
+func Write(c *chart.Chart, w io.Writer) error {
```

Comment:
this diff looks a bit wonky. What's happened here is that a portion of `Save` was extracted into its own function `Write`. That is then used by both `Save` (https://github.com/helm/helm/pull/7613/files#diff-9ea0c476e7eb107c6b0f198af1a49caaR122), and in https://github.com/helm/helm/pull/7613/files#diff-9069fe52469166bbcbe989af4b5ae3f5R80